src/main/java/io/javalin/HttpResponseException.kt

```diff
@@ -8,19 +8,19 @@ package io.javalin
 
 import org.eclipse.jetty.http.HttpStatus
 
-open class HttpResponseException(val status: Int, @Deprecated("Use 'message' instead") val msg: String, val details: Map<String, String> = mapOf()) : RuntimeException(msg)
+open class HttpResponseException(val status: Int, @Deprecated("Use 'message' instead") val msg: String, val details: Map<String, String> = mapOf(), cause: Throwable? = null) : RuntimeException(msg, cause)
 
 class RedirectResponse(status: Int = HttpStatus.FOUND_302, msg: String = "Redirected") : HttpResponseException(status, msg)
 
-class BadRequestResponse(message: String = "Bad request") : HttpResponseException(HttpStatus.BAD_REQUEST_400, message)
+class BadRequestResponse(message: String = "Bad request", cause: Throwable? = null) : HttpResponseException(HttpStatus.BAD_REQUEST_400, message, mapOf(), cause)
 class UnauthorizedResponse(message: String = "Unauthorized") : HttpResponseException(HttpStatus.UNAUTHORIZED_401, message)
 class ForbiddenResponse(message: String = "Forbidden") : HttpResponseException(HttpStatus.FORBIDDEN_403, message)
 class NotFoundResponse(message: String = "Not found") : HttpResponseException(HttpStatus.NOT_FOUND_404, message)
```

Comment:
Use named parameters to avoid having to pass the `details` map.

Reply:
Thx, made it in 2c49f23